src/message.js

```diff
@@ -345,6 +345,15 @@ export class Message {
       enums.read(enums.aead, await getPreferredAlgo('aead', encryptionKeys, date, userIDs, config)) :
       undefined;
 
+    await Promise.all(encryptionKeys.map(key => key.getEncryptionKey()
+      .catch(() => null) // ignore key strength requirements
+      .then(maybeKey => {
+        if (maybeKey && (maybeKey.keyPacket.algorithm === enums.publicKey.x25519) && !util.isAES(algo)) {
+          throw new Error('Could not generate a session key compatible with the given `encryptionKeys`: X22519 keys can only be used to encrypt AES session keys; change `config.preferredSymmetricAlgorithm` accordingly.');
+        }
+      })
+    ));
```

Comment:
I think it would be safer to just use AES128 in this scenario. After all, we're always allowed to use that, and the config is called `preferredSymmetricAlgorithm`, it's not a guarantee it will be used (and in this case we can just say that the key doesn't support it - although tbh it would be a bit strange if the key's algorithm preferences claim otherwise).

Reply:
This is an edge case which requires passing `config.preferredSymmetricAlgorithm` with a non-AES algo as well as (all) encryptionKeys with that algo appearing in the prefs.
So I think a specific error makes sense in this case, also considering that AES-128 is not the best fallback for x488, so users might want to go with AES-256 at that point 🤞

Reply:
Yeah, fair enough :+1: